frame/bridge/ethereum-bsc/src/lib.rs

```diff
@@ -0,0 +1,559 @@
+		/// Difficulty header field is invalid.
+		InvalidDifficulty,
+		/// Non-zero mix digest
+		/// InvalidMixDigest is returned if a block's mix digest is non-zero
+		InvalidMixDigest,
+		/// Header timestamp is ahead of on-chain timestamp
+		HeaderTimestampIsAhead,
+		/// Extra-data 32 byte vanity prefix missing
+		/// MissingVanity is returned if a block's extra-data section is shorter than
+		/// 32 bytes, which is required to store the validator(signer) vanity
+		///
+		/// Extra-data 32 byte vanity prefix missing
+		MissingVanity,
+		/// Extra-data 65 byte signature suffix missing
+		/// MissingSignature is returned if a block's extra-data section doesn't seem
+		/// to contain a 65 byte secp256k1 signature
+		MissingSignature,
+		/// Invalid validator list on checkpoint block
+		/// errInvalidCheckpointValidators is returned if a checkpoint block contains an
+		/// invalid list of validators (i.e. non divisible by 20 bytes)
+		InvalidCheckpointValidators,
+		/// Non-checkpoint block contains extra validator list
+		/// ExtraValidators is returned if non-checkpoint block contain validator data in
+		/// their extra-data fields
+		ExtraValidators,
+
+		/// UnknownAncestor is returned when validating a block requires an ancestor that is unknown.
+		UnknownAncestor,
+		/// Header timestamp too close while header timestamp is too close with parent's
+		HeaderTimestampTooClose,
+
+		/// Missing signers
+		CheckpointNoSigner,
+		/// List of signers is invalid
+		CheckpointInvalidSigners,
+
+		/// EC_RECOVER error
+		///
+		/// Recover pubkey from signature error
+		RecoverPubkeyFail,
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn finalized_authority)]
+	pub type FinalizedAuthority<T> = StorageValue<_, Vec<Address>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn finalized_checkpoint)]
+	pub type FinalizedCheckpoint<T> = StorageValue<_, BSCHeader, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn creator_cache_of)]
+	pub type CreatorCache<T> = StorageMap<_, Identity, Hash, Address, OptionQuery>;
+
+	#[cfg_attr(feature = "std", derive(Default))]
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {
+		pub genesis_header: BSCHeader,
+	}
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			let initial_authority_set =
+				<Pallet<T>>::extract_authorities(&self.genesis_header).unwrap();
+
+			<FinalizedAuthority<T>>::put(initial_authority_set);
+			<FinalizedCheckpoint<T>>::put(&self.genesis_header);
+		}
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Verify unsigned relayed headers and finalize authority set
+		#[pallet::weight(0)]
+		pub fn verify_and_update_authority_set_unsigned(
+			origin: OriginFor<T>,
+			headers: Vec<BSCHeader>,
+		) -> DispatchResultWithPostInfo {
+			// TODO: should we ignore all checks while number equals 0?
```

Comment:
FinalizedCheckpoint should have already been set, and we have checks later for block number. 

We can ignore this.